### ISSUE TITLE
Right click context menu breaks at browser bottom edge

### ISSUE THREAD

**Issue by Sibicle** [jfurrow/flood#652]
## Summary
When right clicking on torrents too close to the bottom of the list, the context menu appears off the top of the view.

## Expected Behavior
When the context menu would appear too close to the bottom of the screen to be practical, it should begin to be nudged upwards relative to the click location, until when the very last list element is pressed, the entire menu is above the click location.

## Current Behavior
If more than four list elements are trying to overflow in the context menu, it is pushed to the top off the browser window, and is mostly obscured

![cursor](https://user-images.githubusercontent.com/4299422/38445266-0e0b5fc2-39b0-11e8-8702-e7d8d11f65cc.png)

## Possible Solution
Should adopt the standard functionality in right-click context menus

## Steps to Reproduce (for bugs)
1. Re-size main window so that there are torrents extending to the bottom of the browser window
2. Right-click the last torrent in the list to bring up its context menu
3. Observe context menu behavior

## Context
Makes the context menu impossible to use for the last few torrents in the list

## Your Environment
* Version used:
  + Version (stable release) `v1.0.0`
  + Commit ID (development release) `293aa11d512ee089f1d559b24e6f8bbe707cfb8d`
* Environment name and version
  + Node `v9.0.0`
  + Chrome `65.0.3325.181 (Official Build) (64-bit)`
  + OS `Debian GNU/Linux 9.4`
* Operating System and version:
  + macOS `10.13.4`

**Comment by noraj** [comment]
Not reproducible with Firefox 59.0.2 64 bits under Manjaro (kernel 4.14.31-1-MANJARO).
Maybe related to #551, and looks exactly the same as #516 (so we may have some regression).

**Comment by noraj** [comment]
@Sibicle Can you try with another browser to see if it is chrome related only. 

For example test with : 

1. Another blink based web browser, ex: Brave, Opera, Vivaldi
2. A browser with another Layout engine, see https://en.wikipedia.org/wiki/List_of_web_browsers

**Comment by Sibicle** [comment]
After testing in Firefox and finding the bug there as well, I became suspicious. After some digging, I realized that `npm build` was failing, because of my particular particular setup on a shared host.

To anyone experiencing problems with updating, make sure you follow the steps outlined in [README.md](https://github.com/jfurrow/flood/blob/master/README.md#updating. Here is what I had to do, due to not having `sudo` access on my server.

In my case flood is installed at `~/flood`, be sure to replace this with your correct install directory.

```
cd ~/flood/
git pull
npm install --production --prefix ~/flood
npm run build --prefix ~/flood
npm start
```

I will add that something was causing problems in my setup, so I actually had to first do:

```
rm -r node_modules
npm install --production --prefix ~/flood
```

before I was able to get an updated version of the app.

Hope this helps!

@noraj, thanks for looking into this, and sorry for the false alarm!
